pkg/events/events.go

```diff
@@ -0,0 +1,122 @@
+package events
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	k8sv1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides struct for Event object which contains connection to cluster.
+type Builder struct {
+	// Dynamically discovered Event object.
+	Objects *k8sv1.Event
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// nsName defines Event namespace.
+	nsName string
+	// nsName defines Event name.
+	eventName string
+	// errorMsg used in discovery function before sending api request to cluster.
+	errorMsg string
+}
+
+// Pull pulls existing Event from cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	glog.V(100).Infof("Pulling existing Event name %s under namespace %s from cluster", name, nsname)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Objects: &k8sv1.Event{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the Event is empty")
+
+		builder.errorMsg = "Event 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the Event is empty")
+
+		builder.errorMsg = "Event 'namespace' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("event object %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	return &builder, nil
+}
+
+// Exists checks whether the given Event exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if Event %s exists", builder.Objects.Name)
+
+	var err error
+	builder.Objects, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns Event object.
+func (builder *Builder) Get() (*k8sv1.Event, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof(
+		"Collecting Event object %s in namespace %s",
+		builder.Objects.Name, builder.Objects.Namespace)
+
+	event := &k8sv1.Event{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
```

Comment:
It looks like you're missing the event namespace in this operation

Reply:
renamed